external-crates/move/crates/move-vm-runtime/src/jit/execution/translate.rs

```diff
@@ -53,11 +48,11 @@ struct PackageContext<'natives> {
 struct FunctionContext<'pkg_ctxt, 'natives> {
     package_context: &'pkg_ctxt PackageContext<'natives>,
     module: &'pkg_ctxt CompiledModule,
-    single_signature_token_map: BTreeMap<SignatureIndex, Type>,
+    single_signature_token_map: BTreeMap<SignatureIndex, ArenaBox<ArenaType>>,
```

Comment:
Trying to figure out why these need to be `ArenaBox`ed types and can't just be `ArenaType`s, probably I'm missing something?

Reply:
`ArenaType` does not live in an `Arena` -- it just contains internal `ArenaVec`/`ArenaType`. We need to place the outer one in an `ArenaBox` or `ArenaVec` to ensure it is well-placed. This is all eliminated in the next PR, though.